Doc restructure

tsavina opened the pull request:
http://openvino-doc.iotg.sclab.intel.com/tsavina_ovms_test/ovms_what_is_openvino_model_server.html
Note that this does not include updated content but only structure and structure-related changes.

dtrawins commented:
The section Starting Model Server is a bit confusing to me. Based on the content it should be called probably "deployment methods". It should have removed building from source (not related) and we should add deployment on baremetal without docker.
Serving models is too generic name while it covers just the server modes topic. Maybe it we should have a section "Starting the model server" including the content from current "Serving models" and "Additional configuration"

mzegla commented:
1. I'm not sure about having parameters described in **Additional configurations** section. Sure, some of them I'd consider additional/advanced, but there are also required parameters that user must know in order to launch OVMS properly. To me the word "additional" hints that I don't really need to look at that to run simple instance. On the other hand we have **Quickstart Guide** and **Serving Models** that show usage with basic parameters, but I thought it's worth considering. 

2. I would move **Model Version Policy** and **Batch, Shape, Layout** to **Additional configurations**.

ngrozae commented on docs/build_from_source.md:
```diff
@@ -0,0 +1,61 @@
+
+cd model_server   
+   
+# automatically build a container from source
+# it places a copy of the binary package in the `dist` subfolder in the Model Server root directory
+make docker_build
+
+# unpack the `tar.gz` file
+cd dist/ubuntu && tar -xzvf ovms.tar.gz
```
```suggestion
# replace to 'redhat` if using UBI base image
export BASE_OS=ubuntu

# automatically build a container from source
# it will also place a copy of the binary package in the `dist` subfolder in the Model Server root directory
make docker_build BASE_OS=${BASE_OS}

# unpack the `tar.gz` file
cd dist/${BASE_OS} && tar -xzvf ovms.tar.gz
```
Can we add BASE_OS env like previously in host.md? Path dist/ubuntu can change depending on OS used

mzegla commented:
@tsavina 

1. I like that idea. We could add a few lines to **Starting Model Server** (or maybe **Serving Models**?) to say that you should know about these required/crucial parameters and to learn about all configuration options see **Advanced Configurations -> Model Server Parameters**.
2. Yeah, I guess some sections could be classified both as configuration and feature, so it depends where we draw the line. I mean, support for cloud storage sounds as much as a feature as model version policy 😄 If we renamed **Model Version Policy** to **Model Versioning** it would sound more as a feature to me. For **Batch, Shape and Layout** I would move it to configuration because it's mostly using OpenVINO capabilities to set model input appropriately - OVMS does not add value there, just passes configuration to OV to adjust model for specific input (except for "auto" option, but it's not recommended with latest versions).

tsavina commented:
> 1. I like that idea. We could add a few lines to **Starting Model Server** (or maybe **Serving Models**?) to say that you should know about these required/crucial parameters and to learn about all configuration options see **Advanced Configurations -> Model Server Parameters**.
> 2. Yeah, I guess some sections could be classified both as configuration and feature, so it depends where we draw the line. I mean, support for cloud storage sounds as much as a feature as model version policy 😄 If we renamed **Model Version Policy** to **Model Versioning** it would sound more as a feature to me. For **Batch, Shape and Layout** I would move it to configuration because it's mostly using OpenVINO capabilities to set model input appropriately - OVMS does not add value there, just passes configuration to OV to adjust model for specific input (except for "auto" option, but it's not recommended with latest versions).

Thanks for your reply @mzegla! I think adding a link to parameters in Serving Models next to the table with required params is a good idea. Updated the PR. 
As for restructuring and drawing the line between features and configurations, I'd like to ask @dtrawins and @ryanloney to comment on this, as they have more technical view on this than I do.

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -1,15 +1,21 @@
+You can add and delete models, as well as update their configurations in runtime, without restarting the model server.
 
-Launch Model Server by running the following command: 
+## Serving Single Model
```
```suggestion
## Serving a Single Model
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -1,15 +1,21 @@
-Launch Model Server by running the following command: 
+## Serving Single Model
+
+Before starting the container, make sure you [prepared the model for serving](models_repository.md).
```
```suggestion
Before starting the container, make sure you have [prepared the model for serving](models_repository.md).
```

ryanloney commented on docs/single_model_mode.md:
````diff
@@ -1,15 +1,21 @@
 ```
 
-Example:
+Example using a resnet model:
````
```suggestion
Example using a ResNet model:
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -20,10 +26,10 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
 
-
-**Configuration Arguments for Running Model Server:**
+The required Model Server parameters are listed below. To learn more about all configuration options, see [Model Server Parameters](parameters.md).
```
```suggestion
The required Model Server parameters are listed below. For additional configuration options, see the [Model Server Parameters](parameters.md) section.
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -48,11 +54,90 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
 
-### Notes
-- Publish the container's port to your host's **open ports**.
+- Publish the container's port to your host's **open ports**. 
```
```suggestion
- Expose the container ports to **open ports** on your host or VM. 
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -48,11 +54,90 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
+
+## Serving Multiple Models 
+
+To use a container with several models, you need an additional JSON configuration file defining each model. In the file, provide a 
```
```suggestion
To serve multiple models from the same container you will need an additional JSON configuration file that defines each model. In the config file, you must provide a 
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -48,11 +54,90 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
+## Serving Multiple Models 
+
+To use a container with several models, you need an additional JSON configuration file defining each model. In the file, provide a 
+`model_config_list` array that includes a collection of config objects for each served model. The `name` and the `base_path` values of the model are required for every config object.
```
```suggestion
`model_config_list` array that includes a collection of config objects for each served model. The `name` and the `base_path` values of the model are required for each config object.
```

ryanloney commented on docs/single_model_mode.md:
````diff
@@ -48,11 +54,90 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
+}
+```
+
+When the Docker container has the config file mounted, it can be started - the command is minimalistic, as arguments are read from the config file. 
````
```suggestion
Once the Docker container has the path to your config file mounted, it can be started. This simplifies the `docker run` command, as arguments are now read from the config file. 
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -48,11 +54,90 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
+
+## Serving Models from Cloud Storage
+
+Note that models with a cloud storage path require setting specific environmental variables. Learn more in [Using cloud storage](using_cloud_storage.md) documentation.
```
```suggestion
Note that models deployed from a cloud storage path require setting specific environmental variables. Learn more in [Using cloud storage](using_cloud_storage.md) documentation.
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -48,11 +54,90 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
+
+## Next Steps
+
+- Try the model server [features](features.md)
```
```suggestion
- Explore all model serving [features](features.md)
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -48,11 +54,90 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
+## Next Steps
+
+- Try the model server [features](features.md)
+- Explore the model server [demos](../demos/README.md)
```
```suggestion
- Try the model serving [demos](../demos/README.md)
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -48,11 +54,90 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
+
+## Additional Resources
+
+- [Preparing Model Repository](models_repository.md)
```
```suggestion
- [Preparing a Model Repository](models_repository.md)
```

ryanloney commented on docs/single_model_mode.md:
```diff
@@ -48,11 +54,90 @@ docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/m
+- [Preparing Model Repository](models_repository.md)
+- [Using Cloud Storage](using_cloud_storage.md)
+- [Troubleshooting](troubleshooting.md)
+- [Model server parameters](parameters.md)
```
```suggestion
- [Model Server Parameters](parameters.md)
```

ryanloney commented on docs/writing_app.md:
```diff
@@ -0,0 +1,23 @@
+# Writing Client Application {#ovms_docs_server_app}
```
```suggestion
# Writing a Client Application {#ovms_docs_server_app}
```

ryanloney commented on docs/writing_app.md:
```diff
@@ -0,0 +1,23 @@
+
+@endsphinxdirective
+
+OpenVINO&trade; Model Server exposes two sets of APIs: one compatible with TensorFlow Serving and another one, with KServe API, for inference. Both APIs work on both gRPC and REST interfaces. Supporting two sets of APIs makes OpenVINO Model Server easier to plug into existing systems the already leverage one of those APIs for inference. Learn more about supported APIs:
```
```suggestion
OpenVINO&trade; Model Server exposes two sets of APIs: one compatible with TensorFlow Serving and another one, with KServe API, for inference. Both APIs work on gRPC and REST interfaces. Supporting two sets of APIs makes OpenVINO Model Server easier to plug into existing systems the already leverage one of these APIs for inference. Learn more about supported APIs:
```

ryanloney reviewed (changes requested):
@tsavina @dtrawins please consider my suggestions

tsavina commented:
> @tsavina @dtrawins please consider my suggestions
@ryanloney thanks for the review, applied

ngrozae commented on docs/deploying_server.md:
```diff
@@ -0,0 +1,108 @@
+result_index = np.argmax(output[0])
+print(imagenet_classes[result_index])' >> predict.py
+
+python predict.py
```
please add expected output  in the next line (zebra) so the output can be checked with tests

tsavina replied:
thanks, added

ngrozae commented on docs/accelerators.md:
````diff
@@ -203,14 +213,14 @@ LATENCY
 ```
 
-THROUGHPUT
+THROUGHTPUT
````
THROUGHPUT was correct

tsavina commented on docs/accelerators.md:
```diff
@@ -203,14 +213,14 @@ LATENCY
             -v ${PWD}/models/public/resnet-50-tf:/opt/model -p 9001:9001 openvino/model_server:latest \
             --model_path /opt/model --model_name resnet --port 9001 \
-            --plugin_config '{"PERFORMANCE_HINT": "THROUGHPUT"}' \
+            --plugin_config '{"PERFORMANCE_HINT": "THROUGHTPUT"}' \
```
Hi, must have fetched someone's changes, rebased to develop version, should be ok now

dtrawins commented on docs/accelerators.md:
````diff
@@ -106,6 +106,31 @@ cd model_server
 make docker_build INSTALL_DRIVER_VERSION=22.10.22597
 ```
 
+## Model Server image with Intel® Data Center GPU Flex Series and Intel® Arc™ GPU support (Ubuntu 20.04)
````
This section is not rebased with develop. new public docker image will include support for Arc and Flex. This section is relevant only to try preproduction driver versions

ngrozae commented on docs/starting_server.md:
````diff
@@ -1,97 +1,144 @@
-## Modifying Model Configuration in Runtime
+docker run -d --rm -v ${PWD}/models:/models -p 9000:9000 -p 9001:9001 openvino/model_server:latest \
+--model_path /models/resnet/ --model_name resnet --port 9000 --rest_port 9001 --log_level DEBUG
+```
````
Can you change rest port number to 8000? This way it will we consistent with the rest of documentation files
